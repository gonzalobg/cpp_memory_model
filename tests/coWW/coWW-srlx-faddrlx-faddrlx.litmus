C coWW-srlx-faddrlx-faddrlx
{ [x] = 0; }

P0 (int* x) {
  atomic_store_explicit(x, 1, memory_order_relaxed);
  atomic_fetch_add_explicit(x, 1, memory_order_relaxed);
}


P1 (int* x) {
  atomic_fetch_add_explicit(x, 0, memory_order_relaxed);
}

~exists ([x]=0 \/ [x]=1)