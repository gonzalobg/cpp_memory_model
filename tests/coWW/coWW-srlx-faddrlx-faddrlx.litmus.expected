Test coWW-srlx-faddrlx-faddrlx Forbidden
States 1
[x]=2;
Ok
Witnesses
Positive: 3 Negative: 0
Condition ~exists ([x]=0 \/ [x]=1)
Observation coWW-srlx-faddrlx-faddrlx Never 0 3
Hash=cbc57f2d3169ae9af3cdf318872f9b8e

